Source/SwiftLintFramework/Models/Example.swift

```diff
@@ -3,22 +3,36 @@
 public struct Example {
     /// The contents of the example
     public private(set) var code: String
+    /// The untyped configuration to apply to the rule, if deviating from the default configuration
+    public private(set) var configuration: Any?
+    /// Whether the example should be tested by prepending multibyte grapheme clusters
+    public private(set) var testMultiByteOffsets: Bool
```

Comment:
It's not very clear from the comment what is actually meant. After reading it, I have several questions that I'd prefer not to have to go digging through the source code to answer:

- why do multibyte grapheme clusters matter?
- why might I want to disable them?
- which multibyte clusters are tested? Are they specific or random?
- where and when are the multibyte characters prepended, and what effect does that have?

Reply:
> why do multibyte grapheme clusters matter?

In the past, some rules were written with some assumptions that each character was one byte.

Characters that extend beyond a single byte encoded in UTF-8 need to be handled correctly.

The `testMultiByteOffsets` setting prepends some multi-byte grapheme clusters to the code in the example (specifically `/* 👨‍👩‍👧‍👦👨‍👩‍👧‍👦👨‍👩‍👧‍👦 */`) to ensure that the locations where violations are reported are correct, and that corrections are properly applied.

> why might I want to disable them?

Some examples aren't resilient to prepending this content to its code and need to disable this part of the test. For example, this unused import code:

```swift
import Foundation
typealias Foo = CFData
```

Gets corrected by replacing `import Foundation` with `import CoreFoundation`:

```swift
import CoreFoundation
typealias Foo = CFData
```

However, when the violating example gets translated to this by the `testMultiByteOffsets` setting:

```swift
/* 👨‍👩‍👧‍👦👨‍👩‍👧‍👦👨‍👩‍👧‍👦 */
import Foundation
typealias Foo = CFData
```

then the correction becomes:

```swift
import CoreFoundation
/* 👨‍👩‍👧‍👦👨‍👩‍👧‍👦👨‍👩‍👧‍👦 */
typealias Foo = CFData
```

Which fails the test. There's nothing wrong with the rule's behavior, it just doesn't match what the simplistic auto-generated rule expects.

> which multibyte clusters are tested? Are they specific or random?

Specific: `/* 👨‍👩‍👧‍👦👨‍👩‍👧‍👦👨‍👩‍👧‍👦 */`

> where and when are the multibyte characters prepended, and what effect does that have?

See above.